[Dx 963] Add do_not_track Analytics FAQ (#4032)
TykTechnologies/tyk-docs: tyk-docs/content/developer-support/frequently-asked-questions/when_to_use_do_not_track.md

---

> title: "When To Use do_not_track Analytics"

Reviewer: Would "do not track" look better?

Reviewer: or maybe JTBD kind of a way. - "why you should not record request analytics"

Author: updated

date: 2024-01-22
tags: ["do_not_track", Analytics]
description: "Explains do_not_track Analytics and the circumstances under which it should be used"
---

> Tyk provides the *do_not_track* configuration parameter to disable tracking analytics at two levels:

Reviewer: ```suggestion
Tyk provides the [do_not_track]({{<ref "advanced-configuration/transform-traffic/endpoint-designer/#do-not-track-endpoint">}}) configuration parameter to disable tracking analytics at two levels:
```

> - **Global**: Analytics are not tracked for all endpoints of an API.

Reviewer: ```suggestion
- **Global**: Analytics are not tracked for all endpoints of an API.
```

> - **Global**: Analytics are not tracked for all endpoints of an API.

Reviewer: We have to be careful with the term "global". reading it I wasn't sure if it was per API or across all APIs.

Author: updated

Author: @letzya text was refactored for code review and improving FAQ since this commit was reviewed. I've restored `do_not_track` in a [later commit](https://github.com/TykTechnologies/tyk-docs/pull/4032/commits/b2bdf9b92a71b7bef7b7b5296da2b6a9bd8af008) now

> - **Global**: Analytics are not tracked for all endpoints of an API.

Reviewer: ```suggestion
- **Per API**: Analytics are not tracked for all endpoints of an API.
```

Reviewer: I think it's clearer to use the term "record" over "track" to explain the meaning

> - **Endpoint**: Analytics can be disabled for selected endpoints.

Reviewer: ```suggestion
- **Endpoint**: Tyk Gateway will not record Analytics for requests to the selected endpoint. Other endpoints will be
tracked as normal.
```

> This can help reduce analytics traffic for systems that are under high load, exhibiting high traffic, e.g. social media platforms (Facebook, Instagram and Twitter), streaming services (YouTube, Netlix and Hulu) and Financial Services and Trading Platforms (PayPal and E*TRADE).

Reviewer: is this a good start to the paragraph `This can help`?

Reviewer: The issue was observed when a user had a high load of CPU. So users need the FAQ to fix a problem and that's the problem. THey wouldn't know to connect the title or the content to that problem. 
So there are 2 problems - 
1. high load of CPU (we can add an image)
2. RPS is not high enough

First problem can be observed, second is more improvement or benefit than a problem. 

So to summaries, I think FAQ needs to start with the problem the user has

Author: updated

Application decisions need to made concerning which endpoints are non critical can thus have analytics disabled. Furthermore, benchmarking and testing will be required to evaluate the actual benefits for the application specific use case.

In the image below we can see two APIs were setup within Tyk Dashboard for conducting a simple load test, to determine the total requests per second for each API:

- **track**: Analytics are tracked for an API, i.e. *track_enabled* is false.
- **notrack**: Analytics are not tracked for an API, i.e. *track_enabled* is true.

> {{< img src="img/faq/do-not-track-usage-scenario/dashboard_apis_measured.png" alt="apis measured in Tyk Dashboard" >}}

Reviewer: ```suggestion
{{< img src="img/faq/do-not-track-usage-scenario/dashboard_apis_measured.png" alt="apis measured in Tyk Dashboard" width="864" >}}
```

Reviewer: add to show you the format of the size

100,000 requests were sent to each API and the total number of requests per second was measured. The results for the *tracked* API are displayed in the left pane terminal window; with the right pane showing the results for the *untracked* API.

{{< img src="img/faq/do-not-track-usage-scenario/do_not_track_performance_impact.png" alt="measuring do_not_track API performance impact" >}}

> We can see that **19253.75** requests per second was recorded for the *untracked* API; with **16743.6011** requests per second reported for the *tracked* API. The number of requests per seconds decreased by **13.0372%** when analytics was enabled. Subsequently, it is worthwhile monitoring traffic and system load and using this feature to improve performance. 

Reviewer: ```suggestion
We can see that **19,253.75** RPS was recorded for the *untracked* API; with **16,743.6011** RPS reported for the *tracked* API. The number of requests per second decreased by **13.0372%** when analytics was enabled. Subsequently, it is worthwhile monitoring traffic and system load and using this feature to improve performance. 
```
